File: GanttChartLightLibraryDemos/Demos/Samples/WPF-CSharp/GanttChartDataGrid/DependencyLineTemplating/PredecessorItemExtensions.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows;

namespace Demos.WPF.CSharp.GanttChartDataGrid.DependencyLineTemplating
{
    public static class PredecessorItemExtensions
    {
        public static readonly DependencyProperty IsImportantProperty = DependencyProperty.RegisterAttached("IsImportant", typeof(bool), typeof(PredecessorItemExtensions));
        public static bool GetIsImportant(DependencyObject obj)
        {
            return (bool)obj.GetValue(IsImportantProperty);
        }
        public static void SetIsImportant(DependencyObject obj, bool value)
        {
            obj.SetValue(IsImportantProperty, value);
        }
    }
}

File: GanttChartLightLibraryDemos/Demos/Samples/WPF-VisualBasic/GanttChartDataGrid/DependencyLineTemplating/PredecessorItemExtensions.vb
Imports System.Text

Public NotInheritable Class PredecessorItemExtensions

    Private Sub New()
    End Sub

    Public Shared ReadOnly IsImportantProperty As DependencyProperty = DependencyProperty.RegisterAttached("IsImportant", GetType(Boolean), GetType(PredecessorItemExtensions))
    Public Shared Function GetIsImportant(obj As DependencyObject) As Boolean
        Return CBool(obj.GetValue(IsImportantProperty))
    End Function
    Public Shared Sub SetIsImportant(obj As DependencyObject, value As Boolean)
        obj.SetValue(IsImportantProperty, value)
    End Sub
End Class
